src/base/solvers/changeCobraSolver.m

```diff
@@ -266,8 +266,20 @@
 % Attempt to set the user provided solver for all optimization problem types
 if strcmpi(solverType, 'all')
     for i = 1:length(OPT_PROB_TYPES)
-        changeCobraSolver(solverName, OPT_PROB_TYPES{i}, 0);
-        fprintf([' > CBT_', OPT_PROB_TYPES{i}, '_SOLVER has been set to ', solverName, '.\n']);
+        try
```

Comment:
There is actually a structure called `SOLVERS` that holds the type of problem a solver can solved.
To access it you can do:
```
global SOLVERS
SOLVERS.glpk.type
```
so the `for` loop could be replaced by: `for i = 1:length(SOLVERS.(solverName).type)` and the `try ...catch` is not needed anymore.